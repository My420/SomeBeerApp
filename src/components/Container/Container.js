import React from 'react';
import PropTypes from 'prop-types';
import './Container.scss';

const Container = ({ children, className }) => {
  return <div className={className}>{children}</div>;
};

Container.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
};

Container.defaultProps = {
  children: null,
  className: 'container'
};

export default Container;
