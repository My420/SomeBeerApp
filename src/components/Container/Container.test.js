import React from 'react';
import { shallow } from 'enzyme';
import Container from './Container';

describe('Test <Container /> component', () => {
  const wrapper = shallow(
    <Container>
      <p>test</p>
    </Container>
  );

  test('should correct render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render tag <div> with class 'container'", () => {
    expect(wrapper.find('div.container')).toHaveLength(1);
  });

  test("should render tag <p> inside <div> with class 'container'", () => {
    expect(wrapper.find('div.container').find('p')).toHaveLength(1);
    expect(
      wrapper
        .find('div.container')
        .find('p')
        .text()
    ).toBe('test');
  });
});
