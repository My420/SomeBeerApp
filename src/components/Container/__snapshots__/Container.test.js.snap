// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Container /> component should correct render component 1`] = `
<div
  className="container"
>
  <p>
    test
  </p>
</div>
`;
